File: flightoptimizer2024_imad/Angular Flight App/FlightOptimizer/src/app/Models/booking.ts
import { Passenger } from "./passenger";
import { Family } from "./family";
import { Flight } from "./flight";
import { FlightSeat } from "./flight-seat";

export class Booking {
    id?:number;
    Reserver?:Passenger;
    Flight?:Flight;
    Family?:Family ;
    Canceled?:boolean=false;
    Seats?:FlightSeat[];
    flightId?:number;
}

File: flightoptimizer2024_imad/OAuth Project/Models/Booking.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace OAuth_Project.Models
{
    public class Booking
    {
        public int Id { get; set; }
        public int? ReserverId {  get; set; }
        public Passenger? Reserver { get; set; }
        public Flight? Flight { get; set; }
        public Family? Family { get; set; }
        public int? FamilyId { get; set; }
        [ForeignKey("Flight")]
        public int FlightId { get;set; }
        public bool Canceled { get; set; } = false;
        public List<FlightSeat>? ReservedSeats { get; set; }
        public float? totalPrice { get; set; } = 0;
    }
}
